src/app/modules/common/layouts/admin-layout/admin-layout.component.ts

```diff
@@ -86,85 +84,87 @@ export class AdminLayoutComponent implements OnInit, AfterViewChecked, AfterView
       if (routeChange instanceof NavigationStart) {
         this.headerPortalOutlet = null;
       }
-      if (routeChange instanceof NavigationEnd && this.isMobile) {
+      if (routeChange instanceof NavigationEnd && this.isMobileView) {
         this.sideNav.close();
       }
     });
+  }
 
-    // Watches screen size and open/close sidenav
-    this.media.media$.pipe(untilDestroyed(this)).subscribe((change: MediaChange) => {
-      this.isMobile = this.layoutService.isMobile;
-      this.updateSidenav();
-      core.emit({ name: 'MediaChange', data: change, sender: this });
-    });
+  detectDeviceType(): void {
+    this.breakpointObserver
+      .observe([Breakpoints.XSmall, Breakpoints.Small])
+      .pipe(untilDestroyed(this))
+      .subscribe((state: BreakpointState) => {
+        this.isMobileView = state.matches;
+        this.updateSidenav();
+        this.cdr.markForCheck();
+      });
+  }
 
+  getStoreUpdates(): void {
     this.store$.pipe(waitForSystemInfo, untilDestroyed(this)).subscribe((sysInfo) => {
       this.hostname = sysInfo.hostname;
     });
 
-    this.core.register({
-      observerClass: this,
-      eventName: 'ForceSidenav',
-    }).pipe(untilDestroyed(this)).subscribe((evt: ForceSidenavEvent) => {
-      this.updateSidenav(evt.data);
-    });
-
-    this.core.register({
-      observerClass: this,
-      eventName: 'SidenavStatus',
-    }).pipe(untilDestroyed(this)).subscribe((evt: SidenavStatusEvent) => {
-      this.isSidenavOpen = evt.data.isOpen;
-      this.sidenavMode = evt.data.mode;
-      this.isSidenavCollapsed = evt.data.isCollapsed;
-    });
-  }
-
-  ngOnInit(): void {
-    this.themes = this.themeService.allThemes;
-    this.currentTheme = this.themeService.currentTheme().name;
-    const navigationHold = document.getElementById('scroll-area');
-
-    // Allows for one-page-at-a-time scrolling in sidenav on Windows
-    if (window.navigator.platform.toLowerCase() === 'win32') {
-      navigationHold.addEventListener('wheel', (event) => {
-        // deltaY is 1 for page scrolling and 33.3 per line for regular scrolling; default is 100, or 3 lines at a time
-        if (event.deltaY === 1 || event.deltaY === -1) {
-          navigationHold.scrollBy(0, event.deltaY * window.innerHeight);
-        }
-      });
-    }
-
-    if (this.media.isActive('xs') || this.media.isActive('sm')) {
-      this.isSidenavOpen = false;
-    }
-    this.sysGeneralService.toggleSentryInit();
-
     this.store$.pipe(waitForGeneralConfig, untilDestroyed(this)).subscribe((config) => {
       this.onShowConsoleFooterBar(config.ui_consolemsg);
     });
 
-    this.isSidenavCollapsed = this.layoutService.isMenuCollapsed;
-
-    this.ws.authStatus.pipe(untilDestroyed(this)).subscribe((evt) => {
-      this.core.emit({ name: 'Authenticated', data: evt, sender: this });
+    combineLatest([
+      this.store$.pipe(waitForGeneralConfig),
+      this.store$.pipe(waitForPreferences),
+    ]).pipe(
+      timeout(10000),
```

Comment:
This seems like an arbitrary amount of seconds. Request could just take longer than this. We shouldn't have to cater the requests specifically to adjust to the new loading screen. Plus, the new loading screen is needlessly blocking the user from content that they could be viewing sooner just because the prefs haven't loaded yet. I think we should remove this screen. @undsoft thoughts?

Reply:
Yeah, it seems better to keep as is